src/sandstorm/run-bundle.c++

```diff
@@ -1796,13 +1796,18 @@ private:
       dropPrivs(config.uids);
       clearSignalMask();
 
-      capnp::TwoPartyServer server(kj::heap<BackendImpl>(*io.lowLevelProvider, network));
+      auto redirector = kj::refcounted<CapRedirector>();
```

Comment:
It seems the reason you need to create the redirector yourself and then pass it in to `TwoPartyServerWithBootstrap` is because of a cyclic problem where TPSWB requires the server's bootstrap as a constructor input, which requires the redirector as an input. But it would be kind of nice if TPSWB could fully encapsulate the concept of the CapRedirector, and only have the one-argument constructor.

A way to solve this would be to use promises. Create a PromiseAndFulfiller<Backend::Client>, pass that as the server's bootstrap, and then immediately fulfill the promise by constructing BackendImpl based on the client's bootstrap from TPSWB::bootstrap().

[Here's an example of something like this from Blackrock.](https://github.com/sandstorm-io/blackrock/blob/16a82a53080c424caa2a743b83f9d9ff8703b143/src/blackrock/blackrock.c++#L507-L521)